code_regen/README.md

```diff
@@ -0,0 +1,84 @@
+goa Code Regeneration
+=====================
+
+Note: This depends on a
+[PR in progress](https://github.com/goadesign/goa/pull/593) for changing TDB to the tag pairs.
+
+---
+
+This example shows how to regenerate your apps code
+without disrupting the modifications you have made.
+This allows you to modify your design, after the initial generation,
+and keep the logic you have already implemented.
+
+__All you have to do is drop the `Makefile` and `restore.py` into your app dirrectory__
+
+and type `make regen`
+
+The process that happens is:
+
+1. Backup the current go files
+1. Remove and regenerate the app
+1. Restore the modifications from the backups
+
+This process is enabled by having [start/end] tags.
+These are genreated automatically for the controller stubs.
+You may also use two special tags `import` and `extra`.
+`import` will appear as the first block in the import clause
+and enables the maintenance of additional imports.
+`extra` will be appended to the end of the file
+and allows extra functions to be added to the file.
+
+
+```Go
+package main
+
+import (
+  // import start_implement
+  "fmt"
```

Comment:
are the import tags really needed? imports change rarely and when they do most people use `goimport` so it doesn't matter anyway. Would be good to keep the script to a minimum.